tools/test/examples/examples_lib.py

```diff
@@ -380,31 +382,60 @@ def compile_repos(config, toolchains, targets, profile, verbose, examples):
         successes = []
         compiled = True
         pass_status = True
+        if example.has_key('test') and example.has_key('baud_rate') and example.has_key('compare_log'):
```

Comment:
This is passing for this kind of json:
{
 "name": "Something",
 "test": False,
 "baund_rate": "invalid",
 "compare_log": null
}

Why it need to have "test" key even if the value is not documented or does not even matter?

These values really need to be documented. Actually same problem with whole examples.json. @adbridge do you know any documentation for that json file?

Reply:
No not that i am aware of....

Reply:
To me, the `test` key is just a on/off switch. similar to the `compile` key, just following the conventions. 
It can be used, when we have the `compare_log` and `baud_rate` ready, but choose not run the test for some reasons. (e.g. the example requires some special HW, but that's not available on the RaaS, or there are some breaking change happening, we need to turn off tests for some examples temperately )

Reply:
@jamesbeyond then the condition need to check the value of example['test'] alse. For example:
`if example.has_key('test') and example['test'] and example.has_key('baud_rate') and example.has_key('compare_log'):`

Reply:
Yes, I was checking that condition in later code, after reviewing, I think what you suggested would be more make scene. So this is fixed